### ISSUE TITLE
Sort search results

### ISSUE THREAD

**Issue by jeberger** [actionless/pikaur#475]
```
> pikaur -Vq
Pikaur v1.6.9.1
Pacman v5.2.1 - libalpm v12.0.1
```

When running `pikaur -Su`, I can use the `UpgradeSorting` option to choose how packages are sorted. But when running `pikaur -Ss` or  `pikaur foo`, packages are sorted more or less randomly (*). It would be nice if the results could be sorted alphabetically inside each repo.

(*) In fact it looks like the results are sorted by reverse (!) alphabetical order for normal repos and randomly for AUR.

**Comment by naufik** [comment]
Hello, first time trying to contribute here. Just had a look at the codebase and I will attempt to do this ticket. Might take 1-2 weeks due to time constraints :)

**Comment by actionless** [comment]
@naufik no worries, i don't have plans to change related code soon, so it shouldn't be possible merge conflicts

**Comment by naufik** [comment]
Almost done implementing. Right now, I have:

- **For official packages**: sort by repo or pkgname
- **For aur packages**: old style sorting[1], pkgname, lastmodified, numvotes, popularity

@actionless @jeberger, any suggestions or requests for other keys to sort by? I don't think `versiondiff` is sensible on non-sync operations - and `lastmodified` is not as useful in official repos (on AUR, I can see a use case where we can use last modified to guess whether a package is likely to be outdated or not).

[1] apparently the "random" sorting for aur was sorted by a metric weighted by popularity and numvotes, I kept this as `pikascore` as I couldn't find a more appropriate name for it. Suggestions welcome :slightly_smiling_face:

**Comment by actionless** [comment]
mb it could be named the same as similar metric named on reddit -- "hottest" packages

**Comment by actionless** [comment]
fixed in #522
